### PR TITLE
[formrecognizer] Update to use Python 3 only

### LINKED ISSUE #22088: [formrecognizer] drop support for 2.7
See: https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/packaging.md

Also update README, samples README, and any other FR-specific docs where we might have included support python versions.

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes #22088

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt [hunk 1]
```diff
@@ -3,4 +3,4 @@
 -e ../../cognitiveservices/azure-mgmt-cognitiveservices
 ../../core/azure-core
 -e ../../identity/azure-identity
-aiohttp>=3.0; python_version >= '3.6'
\ No newline at end of file
+aiohttp>=3.0; python_version >= '3.7'
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt:6]: Now that I look at this again, I think we can remove the conditional, but keep the >=3.0

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt (hunk 1 above)
Comment (lmazuel) [reply sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt:6]: I just checked with @scbedd and this should be fine, good catch

Comment (lmazuel) [reply sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt:6]: Though it's an unecessary test, it's not having an bad impact, so not critical

**Review by lmazuel — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/dev_requirements.txt:6]: Ok will leave as is for now. Thanks for the reviews!

**Pull request merged**